examples/go/vm-reconfig-via-tag/handler/handler.go

```diff
@@ -0,0 +1,564 @@
+			vsc.rest.SessionID(),
+		)
+	}
+
+	return &vsc, nil
+}
+
+func handleSignal(ctx context.Context, vsc *vsClient) {
+	sigCh := make(chan os.Signal, 2)
+
+	signal.Notify(sigCh, syscall.SIGTERM, os.Interrupt)
+
+	if debug {
+		log.Println("Started signal handling.")
+	}
+
+	s := <-sigCh
+
+	err := vsc.govmomi.Logout(ctx)
+	if err != nil {
+		log.Printf("log out of Govmomi: %v", err)
+	}
+
+	err = vsc.rest.Logout(ctx)
+	if err != nil {
+		log.Printf("log out of VAPI: %v", err)
+	}
+
+	if debug {
+		log.Printf("got signal: %v, logged out of vSphere APIs", s)
+	}
+}
+
+func (c *vsClient) attachedTags(ctx context.Context, mor types.ManagedObjectReference) ([]tags.Tag, error) {
+	// Tag objects attached to the VM.
+	tagObjs, err := c.tagMgr.GetAttachedTags(ctx, mor)
+	if err != nil {
+		return []tags.Tag{}, fmt.Errorf("get attached tags: %w", err)
+	}
+
+	if debug {
+		log.Println("Retrieved", len(tagObjs), "tag(s).")
+	}
+
+	return tagObjs, nil
+}
+
+func (c *vsClient) moVirtualMachine(ctx context.Context, mor types.ManagedObjectReference) (mo.VirtualMachine, error) {
+	var moVM mo.VirtualMachine
+	pc := property.DefaultCollector(c.govmomi.Client)
+
+	err := pc.Retrieve(ctx, []types.ManagedObjectReference{mor}, []string{}, &moVM)
+	if err != nil {
+		return mo.VirtualMachine{}, fmt.Errorf("retrieve VM configurations :%w", err)
+	}
+
+	if debug {
+		log.Println("Retrieve VM configs", moVM)
+	}
+
+	return moVM, nil
+}
+
+// currentConfigs gets a list of current selected configs set for the VM object.
+func currentConfigs(ctx context.Context, moVM mo.VirtualMachine) []vmConfig {
+	configs := make([]vmConfig, 6)
+
+	configs[0] = vmConfig{name: "numCPU", valInt: int(moVM.Config.Hardware.NumCPU)}
+	configs[1] = vmConfig{name: "memoryMB", valInt: int(moVM.Config.Hardware.MemoryMB)}
+	configs[2] = vmConfig{name: "numCoresPerSocket", valInt: int(moVM.Config.Hardware.NumCoresPerSocket)}
+
+	if moVM.Config.MemoryHotAddEnabled != nil {
+		configs[3] = vmConfig{name: "memoryHotAddEnabled", valBool: *moVM.Config.MemoryHotAddEnabled}
+	}
+
+	if moVM.Config.CpuHotRemoveEnabled != nil {
+		configs[4] = vmConfig{name: "cpuHotRemoveEnabled", valBool: *moVM.Config.CpuHotRemoveEnabled}
+	}
+
+	if moVM.Config.CpuHotAddEnabled != nil {
+		configs[5] = vmConfig{name: "cpuHotRemoveEnabled", valBool: *moVM.Config.CpuHotAddEnabled}
+	}
+
+	if debug {
+		log.Println("Current VM configurations:", configs)
+	}
+
+	return configs
+}
+
+// selectConfigs determines which tagged configurations and their values
+// that need to be applied.
+func (c *vsClient) unappliedConfigs(ctx context.Context, tagObjs []tags.Tag, currCfgs []vmConfig) ([]vmConfig, error) {
+	// Get config information from tags.
+	tagCfgs, err := c.desiredConfigs(ctx, tagObjs)
+	if err != nil {
+		return []vmConfig{}, fmt.Errorf("get config info from tags: %w", err)
+	}
+
+	unappliedCfgs := filterOutCurrentConfigs(tagCfgs, currCfgs)
+
+	if debug {
+		log.Println("List of unapplied configs:", unappliedCfgs)
```

Comment:
Missing spaces between strings.

Reply:
Println adds spaces in between the arguments.